### PR TITLE
Feature/cplp 2061/privacy policy

### PR DESCRIPTION
**Pull request by lavanya-bmw**

### CODE REVIEW

File: cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 1]
```diff
@@ -556,6 +586,49 @@ export default function AppPage() {
...
+              ))}
+            <Grid item md={6}>
+              <Radio
+                label={
+                  getPrivacyPolicies &&
+                  getPrivacyPolicies?.privacyPolicies.slice(-1)[0]
+                }
+                checked={
+                  getValuesPrivacyPolicies.length > 0 &&
+                  getValuesPrivacyPolicies[0] === 'NONE'
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:619]: Replace `NONE` with `PrivacyPolicyType.NONE`

**Review by oyo — CHANGES_REQUESTED** [review]
Please replace the privacy policy string everywhere with the existing PrivacyPolicyType.
https://github.com/catenax-ng/tx-portal-frontend/blob/dev/cx-portal/src/features/adminBoard/adminBoardApiSlice.ts#L29-L35

When we have return types that come as string from API but are not arbitrary please define and use enum types like here:

```export enum PrivacyPolicyType {
  COMPANY_DATA = 'COMPANY_DATA',
  USER_DATA = 'USER_DATA',
  LOCATION = 'LOCATION',
  BROWSER_HISTORY = 'BROWSER_HISTORY',
  NONE = 'NONE',
}```

this will reduce possible errors in case of a typo and also help developing because the IDE will suggest the possible values

File: cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 2]
```diff
@@ -350,6 +366,20 @@ export default function AppPage() {
     dispatch(decrement())
   }
 
+  const selectPrivacyPolicies = (policy: string, select: boolean) => {
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:369]: `string` -> `PrivacyPolicyType

File: cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 3]
```diff
@@ -64,6 +68,7 @@ type FormDataType = {
   providerHomePage: string
   providerContactEmail: string
   providerPhoneContact: string
+  privacyPolicies: string[] | []
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:71]: `string` -> `PrivacyPolicyType`

**Comment by lavanya-bmw** [comment]
> Please replace the privacy policy `string` everywhere with the existing `PrivacyPolicyType`. https://github.com/catenax-ng/tx-portal-frontend/blob/dev/cx-portal/src/features/adminBoard/adminBoardApiSlice.ts#L29-L35
> 
> I haven't commented on each occurrence. As a general rule when we have return types coming as string from API that are not arbitrary but a distinct set of values please define and use enum types in the apiSlice like here:
> 
> ```
> export enum PrivacyPolicyType {
>   COMPANY_DATA = 'COMPANY_DATA',
>   USER_DATA = 'USER_DATA',
>   LOCATION = 'LOCATION',
>   BROWSER_HISTORY = 'BROWSER_HISTORY',
>   NONE = 'NONE',
> }
> ```
> 
> this will reduce possible errors in case of a typo and also help developing because the IDE will suggest the possible values

As discussed, since the privacy policies are fetched from API, I'm not defining static enums.

**Review by oyo — APPROVED** [review]

**Pull request merged**